web/src/drawing/layoutFormula.ts

```diff
@@ -76,88 +75,94 @@ function getBoundingRec(
+
+    if (
+      !checkOutcomeAgainstViewingFilters(
+        outcomes[connection.parentActionHash],
+        hiddenSmalls,
+        hiddenAchieved
+      ) || // parent is hidden
+      !checkOutcomeAgainstViewingFilters(
+        outcomes[connection.childActionHash],
+        hiddenSmalls,
+        hiddenAchieved
+      ) || // child is hidden
+      projectCollapsedOutcomes[connection.parentActionHash] || // parent is collapsed
+      projectCollapsedOutcomes[connection.childActionHash] // child is collapsed
+    ) {
+      return
+    }
+
+    nodeConnections.push([
+      connection.parentActionHash,
+      connection.childActionHash,
+    ])
   })
 
-  const layoutTree = {}
-  // use recursion to add each outcome as a node in the graph
-  function addOutcome(outcome: ComputedOutcome, node: any, level: number) {
-    let numDescendants = 0
-    node.children = []
-    // Skip over the Outcome if it is included in the list
-    // of collapsed Outcomes
-    if (!projectCollapsedOutcomes[outcome.actionHash]) {
-      outcome.children.forEach((childOutcome) => {
-        const childNode = {}
-        // filter out children who don't match the viewing filter criteria
-        if (
-          checkOutcomeAgainstViewingFilters(
-            childOutcome,
-            hiddenSmalls,
-            hiddenAchieved
-          )
-        ) {
-          const descendantCount = addOutcome(childOutcome, childNode, level + 1)
-          node.children.push(childNode)
-          numDescendants += descendantCount + 1
-        }
-      })
+  // add all the nodes without connections
+  Object.keys(outcomes).forEach((hash) => {
+    const outcome = outcomes[hash]
+    if (
+      !nodeConnections.find(
+        (nodeConnection) =>
+          nodeConnection[0] === outcome.actionHash ||
+          nodeConnection[1] === outcome.actionHash
+      ) &&
+      checkOutcomeAgainstViewingFilters(outcome, hiddenSmalls, hiddenAchieved)
+    ) {
+      nodeConnections.push([outcome.actionHash, outcome.actionHash])
     }
+  })
 
-    const width = allOutcomeDimensions[outcome.actionHash].width
-    // to create the dynamic vertical height, we apply
-    // a power law to the number of descendants, with a heightened baseline
-    // because the number of descendants is a good measure of how wide this tree
-    // might be
-    const height =
-      allOutcomeDimensions[outcome.actionHash].height +
-      (numDescendants > 0 ? Math.pow(numDescendants + 25, 1.5) : 60) +
-      VERTICAL_SPACING
+  // if there are no nodes, return an empty object
+  if (!nodeConnections.length) return {}
 
-    node.actionHash = outcome.actionHash
-    node.size = [width, height]
+  const create = connect().single(true) // allow single nodes without connections
+  const dag = create(nodeConnections as any)
+  const dagLayout = sugiyama()
+    // NOTE: make the width and height dynamic based on the card dimensions
+    .nodeSize((node) => (node === undefined ? [0, 0] : [750, 500]))
```

Comment:
we need to get these width and height to be dynamic, so that the layout is more optimized for space. 
These nodes would tighten up next to each other horizontally.
<img width="535" alt="Screenshot 2023-04-21 at 4 28 28 PM" src="https://user-images.githubusercontent.com/1409121/233748595-fec285ef-2f00-4721-9574-a9bf0b2fdbc4.png">